src/context/virtual-drive/files/application/FileFolderContainerDetector.ts

```diff
@@ -0,0 +1,18 @@
+import { FolderFinder } from '../../folders/application/FolderFinder';
+import { File } from '../../files/domain/File';
+import { Folder } from '../../folders/domain/Folder';
+import { FileRepository } from '../domain/FileRepository';
+
+export class FileFolderContainerDetector {
+  constructor(
+    private readonly repository: FileRepository,
+    private readonly folderFinder: FolderFinder
+  ) {}
+
+  run(contentId: File['contentsId'], folderContentId: Folder['uuid']): boolean {
+    const file = this.repository.searchByPartial({ contentsId: contentId });
+    const folder = this.folderFinder.findFromId(file?.folderId);
```

Comment:
I would avoid having findFromId with the id optional. If there is no ID we already know that it will fail to find it. So I would add a guard clause, this way if the file is not found, then the error thrown is from the file not being in the repo and not that a folder